drivers/platform/mbed/mbed_timer.cpp

```diff
@@ -0,0 +1,191 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/*****************************************************************************/
+/***************************** Include Files *********************************/
+/*****************************************************************************/
+
+#include <stdio.h>
+#include <mbed.h>
+
+//Platform Drivers need to be C-compatible to work with other drivers
+#ifdef __cplusplus
+extern "C"
+{
+#endif	//__cplusplus
+
+#include "no-os/timer.h"
+#include "no-os/error.h"
+#include "mbed_timer.h"
+
+using namespace std::chrono;
+
+/*****************************************************************************/
+/************************** Functions Definitions ****************************/
+/*****************************************************************************/
+
+/**
+ * @brief Initialize the timer peripheral.
+ * @param desc[in, out] - The Timer descriptor.
+ * @param init_param[in] - Structure that contains Timer Initialization Parameters.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t timer_init(struct timer_desc **desc,
+		   struct timer_init_param *param)
+{
+	struct timer_desc *timer_desc; 		// Pointer to timer descriptor
+	struct mbed_timer_desc *mbed_timer_desc;	// Pointer to mbed timer descriptor
+	mbed::Timer *timer;					// Pointer to new Timer instance
+
+	if (!desc || !param)
+		return -EINVAL;
+
+	timer_desc = (struct timer_desc *)calloc(1, sizeof(*timer_desc));
+	if (!timer_desc)
+		return -ENOMEM;
+
+	mbed_timer_desc = (struct mbed_timer_desc *)calloc(1, sizeof(*mbed_timer_desc));
+	if (!mbed_timer_desc)
+		goto err_mbed_timer_desc;
+
+	timer = new Timer();
+	if (!timer)
+		goto err_timer;
+
+	mbed_timer_desc->timer = timer;
+	timer_desc->extra = mbed_timer_desc;
+
+	*desc = timer_desc;
+	return SUCCESS;
+
+err_timer:
+	free(mbed_timer_desc);
+err_mbed_timer_desc:
+	free(timer_desc);
+
+	return -ENOMEM;
+}
+
+/**
+* @brief Start the Timer.
+* @param desc[in] - The Timer Descriptor.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t timer_start(struct timer_desc *desc)
+{
+	mbed::Timer *timer;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	timer = (Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer);
+	if (!timer)
+		return -ENOMEM;
+
+	timer->start();
+
+	return SUCCESS;
+}
+
+/**
+* @brief Stop the Timer.
+* @param desc[in] - The Timer Descriptor.
+* @return SUCCESS in case of success, negative error code otherwise.
+*/
+int32_t timer_stop(struct timer_desc *desc)
+{
+	mbed::Timer *timer;
+
+	if (!desc || !desc->extra)
+		return -EINVAL;
+
+	timer = (Timer *)(((struct mbed_timer_desc *)(desc->extra))->timer);
```

Comment:
ditto as in https://github.com/analogdevicesinc/no-OS/pull/1243/files#r802329865